text-serializer-legacy/src/main/java/net/kyori/adventure/text/serializer/legacy/LegacyComponentSerializer.java

```diff
@@ -261,4 +263,39 @@
     @Override
     @NonNull LegacyComponentSerializer build();
   }
+
+  /**
+   * A {@link LegacyComponentSerializer} service provider.
+   *
+   * @since 4.8.0
+   */
+  @ApiStatus.Internal
+  interface Provider {
+    /**
+     * Provides a {@link LegacyComponentSerializer} using {@link #AMPERSAND_CHAR}.
+     *
+     * @return a {@link LegacyComponentSerializer}
+     * @since 4.8.0
+     */
+    @ApiStatus.Internal
+    @NonNull LegacyComponentSerializer legacyAmpersand();
+
+    /**
+     * Provides a {@link LegacyComponentSerializer} using {@link #SECTION_CHAR}.
+     *
+     * @return a {@link LegacyComponentSerializer}
+     * @since 4.8.0
+     */
+    @ApiStatus.Internal
+    @NonNull LegacyComponentSerializer legacySection();
+
+    /**
+     * Completes the building process of {@link Builder}.
```

Comment:
This is not called on completion, but initialization, unless you mean completion of the `Builder` constructor which is not at all clear.